lib/templates-sync.php

```diff
@@ -106,3 +124,12 @@ function _gutenberg_synchronize_theme_templates( $template_type ) {
 		_gutenberg_create_auto_draft_for_template( $template_post_types[ $template_type ], $slug, wp_get_theme()->get_stylesheet(), $content );
 	}
 }
+
+/**
+ * Synchronize changed template and template part files after WordPress is loaded
+ */
+function gutenberg_synchronize_theme_templates_on_load() {
+	_gutenberg_synchronize_theme_templates( 'template' );
+	_gutenberg_synchronize_theme_templates( 'template_part' );
+}
+add_action( 'wp_loaded', 'gutenberg_synchronize_theme_templates_on_load' );
```

Comment:
Since the `templates-sync.php` file is loaded all the time (regardless of whether an FSE theme is active), we should probably make sure that `gutenberg_synchronize_theme_templates_on_load` only runs if FSE is enabled. (e.g. shortcut if `gutenberg_is_fse_theme` is false)

Reply:
Gotcha! Thanks!